src/Xamarin.Android.Tools.AndroidSdk/Jdks/JdkLocations.MacOS.cs

```diff
@@ -21,8 +21,10 @@ static IEnumerable<string> GetUnixConfiguredJdkPaths (Action<TraceLevel, string>
 		{
 			var config = AndroidSdkUnix.GetUnixConfigFile (logger);
 			foreach (var java_sdk in config.Root.Elements ("java-sdk")) {
-				var path    = (string) java_sdk.Attribute ("path");
-				yield return path;
+				var path    = (string?) java_sdk.Attribute ("path");
+				if (path != null && !string.IsNullOrEmpty (path)) {
```

Comment:
the `path != null` check is redundant, right?

Reply:
You'd *think* so, but… Different .NET profiles have different NRT metadata, and iirc in .NET Standard 2.0, `string.IsNullOrEmpty()` does *not* have `[NotNullWhen(false)]` on the parameter, which results in warnings if it isn't explicitly checked.  :-(

```
JdkLocations.MacOS.cs(26,19): warning CS8603: Possible null reference return.
```